hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/HoodieFlinkCopyOnWriteTable.java

```diff
@@ -431,6 +434,20 @@ public HoodieRestoreMetadata restore(HoodieEngineContext context, String restore
   // -------------------------------------------------------------------------
   //  Used for compaction
   // -------------------------------------------------------------------------
+
+  @Override
+  public List<WriteStatus> compactUsingFileGroupReader(
+      String instantTime,
+      CompactionOperation operation,
+      HoodieWriteConfig writeConfig,
+      HoodieReaderContext readerContext) {
+    config.setDefault(writeConfig);
```

Comment:
not sure why this is needed.

Reply:
Just follows HoodieSparkCopyOnWriteTable#compactUsingFileGroupReader